scaladoc/src/dotty/tools/scaladoc/tasty/TypesSupport.scala

```diff
@@ -272,6 +272,17 @@ trait TypesSupport:
 
       case RecursiveType(tp) => inner(tp)
 
+      case MatchCase(pattern, rhs) =>
+        texts("case ") ++ inner(pattern) ++ texts(" => ") ++ inner(rhs)
+
+      case t: dotty.tools.dotc.core.Types.LazyRef => try {
+        inner(t.ref(using ctx.compilerContext).asInstanceOf[TypeRepr])
+      } catch {
+        case e: AssertionError => texts("LazyRef(...)")
+      }
+
+      case tpe => throw MatchError(tpe.show(using Printer.TypeReprStructure))
```

Comment:
I think we should more verbose message that this is a bug and one needs to report it

Reply:
Done